src/main/java/sparksoniq/spark/iterator/function/ParquetFileFunctionIterator.java

```diff
@@ -54,7 +54,22 @@ public void open(DynamicContext context) {
     }
 
     @Override
-    public JavaRDD<Item> getRDD(DynamicContext context) {
+    public boolean isDataFrame() {
+        return true;
+    }
+
+    @Override
+    public Dataset<Row> getDataFrame(DynamicContext context) {
+        RuntimeIterator urlIterator = this._children.get(0);
+        urlIterator.open(context);
+        Dataset<Row> rows = SparkSessionManager.getInstance().getOrCreateSession()
+                .read().parquet(urlIterator.next().getStringValue());
+        urlIterator.close();
+        return rows;
+    }
+
+    @Override
+    public JavaRDD<Item> getRDDAux(DynamicContext context) {
```

Comment:
This method can now be completely dropped. Indeed, is DataFrame always returns true and getRDDAux will never be called.

This is a general rule: an iterator that always return true to isDataFrame() does not need to provide getRDDAux. an iterator that always return true to isRDD() does not need to provide local iterators.

Only iterators whose answer to isDataFrame() may sometimes be yes, sometimes be no, need to provide getRDDAux.

Does it make sense?